pytorch_lightning/loops/base.py

```diff
@@ -295,21 +295,30 @@ def load_state_dict(
         state_dict: Dict,
         prefix: str = "",
         metrics: Optional[Dict[str, Metric]] = None,
+        force_load_progress: bool = False,
     ) -> None:
         """Loads the state of this loop and all its children."""
-        self._load_from_state_dict(state_dict.copy(), prefix, metrics)
+        self._load_from_state_dict(state_dict.copy(), prefix, metrics, force_load_progress)
         for k, v in self.__dict__.items():
             if isinstance(v, Loop):
-                v.load_state_dict(state_dict.copy(), prefix + k + ".")
+                v.load_state_dict(state_dict.copy(), prefix + k + ".", force_load_progress=force_load_progress)
+
+    def _load_from_state_dict(
+        self,
+        state_dict: Dict,
+        prefix: str,
+        metrics: Optional[Dict[str, Metric]] = None,
+        force_load_progress: bool = False,
+    ) -> None:
+        load_progress = _FaultTolerantMode.detect_current_mode().is_enabled or force_load_progress
 
-    def _load_from_state_dict(self, state_dict: Dict, prefix: str, metrics: Optional[Dict[str, Metric]] = None) -> None:
         for k, v in self.__dict__.items():
             key = prefix + k
             if key not in state_dict:
                 # no state for this object, maybe we are loading an old checkpoint
                 continue
 
-            if isinstance(v, BaseProgress):
+            if load_progress and isinstance(v, BaseProgress):
```

Comment:
Same here?

Reply:
this was done to make sure the progress state is loaded if required else during state_dict it used to save only when fault_tolerant is True, but now it always saves it and while loading back, we have an argument(default=False) that force reloads the base progress state_dict if it's set to True else only reloads it if fault-tolerant is enabled.